sklearn/utils/estimator_checks.py

```diff
@@ -946,10 +956,15 @@ def check_array_api_input(
         "transform",
     )
 
+    methods_to_skip = skips.get(array_namespace, [])
+
     for method_name in methods:
         method = getattr(est, method_name, None)
-        if method is None:
-            continue
+        if method is None or method_name in methods_to_skip:
+            raise SkipTest(
+                f"{array_namespace} is not Array API compliant for method"
+                f" {method_name} of {name}"
+            )
```

Comment:
Why the change for `method is None`? This is the case when an estimator doesn't have a particular method, in which case we aren't skipping a test (and certainly not because of missing Array API support :D)